src/models/ontology_models.py

```diff
@@ -0,0 +1,118 @@
+
+from sentence_transformers import util
+import pandas as pd
+import numpy as np
+from sentence_transformers import SentenceTransformer
+from transformers import AutoTokenizer, AutoModel, AutoModelForMaskedLM
+from tqdm.auto import tqdm
+import torch
+from abc import ABC, abstractmethod
+import yaml
+
+class OntoModelsBase:
+    def __init__(self, method:str, query:list, corpus:list, yaml_path:str='method_model.yaml') -> None:    
```

Comment:
if the argument type is a list or dict, I would specify the type further (list of strings, or list of integers, for example)

Reply:
Aah got it. Updating that now